### DISCUSSION TITLE
Middleware for relations?

### DISCUSSION THREAD

**Discussion by Rykuno** [prisma/prisma#8370] [Q&A] [answered]
**-- Stack --**
- type-graphql
- typedi
- apollo-server-express
- Prisma

**(1)** Is there a method/middleware to automatically apply relations to an operation?

I have multiple areas in which i must manually specify an `include` clause for fetching a model such as the example below, and I'm afraid as the application grows this will become a chore. 
I was wondering if there was some way to simply create middleware on the `User` model to automate this.
```js
    const user = await prisma.user.findUnique({
      where: { id },
      include: { following: true, followedBy: true },
    })
```

**(2)** Also, I'm utilizing `type-graphql` and for my introspection to pickup the `followedBy` and `following` fields, I must explicitly define them within my resolver. I figured since they were included in the model, this step would be redundant; or maybe I'm misunderstanding.

###   Resolver 
```js
  @FieldResolver(() => UsersResponse)
  async followedBy(@Root() user: User) {
    const items = user.followedBy || []
    const count = user.followedBy?.length || 0
    return { items, count }
  }

  @FieldResolver(() => UsersResponse)
  following(@Root() user: User) {
    const items = user.following || []
    const count = user.following?.length || 0
    return { items, count }
  }
  ```
  
###   Model 
```js
model User {
  id          String   @id @default(uuid())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @default(now()) @updatedAt

  followedBy User[] @relation("UserFollows", references: [id])
  following  User[] @relation("UserFollows", references: [id])
}
```

### Current Solution
Currently I have an `@Service` class to my `@Resolver` and make all queries for users through that which includes the two relations I'm expecting upon a return. I'm hoping someone can enlighten me if this is a practice I should be continuing or if there is a better way.

```js
@Service()
export class UserService {
   ...
  async fetchUser(where: UserWhereUniqueInput): Promise<User> {
    const user = await prisma.user.findUnique({
      where,
      include: { following: true, followedBy: true },
    })

    if (!user) throw new UserInputError("User does not exist")
    return user
  }
}
```

**Comment by ryands17** [comment, marked as answer]
@Rykuno 👋 

I think Prisma's [Middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) should help here where you can set the arguments to have `include` present.

Also are you using TypeGraphQL's [Prisma generator](https://typegraphql.com/docs/prisma.html)? If so, then it should automatically generate the resolvers for you with the relations so you don't ned to write them yourself.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
